test/parallel/test-worker-init-failure.js

```diff
@@ -0,0 +1,43 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const child_process = require('child_process');
+
+// Test that workers fail with meaningful error message
+// when their initialization fails.
+
+if (process.argv[2] === 'child') {
+  const { Worker } = require('worker_threads');
+  for (let i = 0; i < 256; ++i) {
+    const worker = new Worker(
+      'require(\'worker_threads\').parentPort.postMessage(2 + 2)',
+      { eval: true });
+    worker.on('message', (result) => {
+      assert.strictEqual(result, 4);
+    });
+    worker.on('error', (e) => {
+      assert(e.message.match(/Worker initialization failure: EMFILE/));
+      assert.strictEqual(e.code, 'ERR_WORKER_INIT_FAILED');
+    });
+  }
+
+} else {
+
+  if (common.isWindows) {
+    common.skip('ulimit does not work on Windows.');
+  }
+
+  // Limit the number of open files, to force workers to fail
+  let testCmd = 'ulimit -n 128 && ';
```

Comment:
Any reason not to go with something much smaller than `128`, like maybe `8`?

Reply:
@Trott, When I ran a simple node.js code, it created 30+ file descriptors. If I reduce the fd count to 8, it fails at arbitrary point without / before reaching worker creation.